angular-workspace/projects/ni/nimble-angular/src/directives/radio/radio-control-value-accessor.directive.ts

```diff
@@ -0,0 +1,297 @@
+/* eslint-disable @angular-eslint/no-host-metadata-property */
+/* eslint-disable @angular-eslint/directive-class-suffix */
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/member-ordering */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-member-accessibility */
+/* eslint-disable max-classes-per-file */
+
+/**
+ * The contents of this file were taken with minimal modification from Angular's source:
+ * https://github.com/angular/angular/blob/0d65e1de2c8ea76a5e48df227638cdeabac82c42/packages/forms/src/directives/ng_control.ts
+ * https://github.com/angular/angular/blob/0d65e1de2c8ea76a5e48df227638cdeabac82c42/packages/forms/src/directives/control_value_accessor.ts
+ * https://github.com/angular/angular/blob/0d65e1de2c8ea76a5e48df227638cdeabac82c42/packages/forms/src/directives/radio_control_value_accessor.ts
+ * See https://github.com/ni/nimble/issues/782 for motivation/context.
+ *
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Directive, ElementRef, forwardRef, Injectable, Injector, Input, NgModule, OnDestroy, OnInit, Renderer2 } from '@angular/core';
+import { AbstractControlDirective, ControlContainer, ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+/**
+ * Angular's NgControl
+ */
+abstract class NgControl extends AbstractControlDirective {
```

Comment:
Are you sure this needs to be forked? It appears public from everything I can see.

Reply:
`RadioControlRegistry` uses `NgControl._parent` which is marked `@internal` in the original file. I guess that causes it to be stripped from the exported declaration, because I get a compile error if I rely on the original export. That's why I had to fork it.

Reply:
I'm hesitant to fork this class because it is a base class and I'm concerned controls may do instanceof checks or use the type for dependency inject, etc. ie all FormControls are expected to extend it:
![image](https://user-images.githubusercontent.com/1588923/196752009-6c60bd2b-da1b-4588-96ab-86e66f970fa0.png)

Because it is marked internal instead of private I think it's name won't be rewritten. We should try and use the same NgControl base class and maybe have a `@ts-ignore` to access the property

Reply:
Now using exported version of NgControl and using `@ts-ignore` to avoid the compile error.